### DISCUSSION TITLE
when called split_video_ffmpeg throw an error

### DISCUSSION THREAD

**Discussion by pengyouak** [Breakthrough/PySceneDetect#454] [Troubleshooting]
Some videos will throw this error halfway through the process

bool' object has no attribute 'shape'

**Comment by Breakthrough** [comment]
Can you provide more context?  A backtrace, etc?  What OS, Python, and package versions you're using?  Have you tried updating OpenCV or using a different backend?

Run `scenedetect version` and report that.  Can you also share more about your workflow?

**Comment by pengyouak** [comment]
[PySceneDetect] PySceneDetect 0.6.4

System Info
------------------------------------------------------------
OS           Windows-10-10.0.22621-SP0
Python       3.8.6

Packages
------------------------------------------------------------
av           Not Installed
click        8.1.7
cv2          4.10.0
moviepy      1.0.3
numpy        1.24.3
platformdirs 4.3.6
scenedetect  0.6.4
tqdm         4.66.5

Tools
------------------------------------------------------------
ffmpeg       4.2.2
mkvmerge     Not Installed

===================================================================================
Output
-------------------
```
 Detected: 29 | Progress: 100%|| 2181/2181 [00:30<00:00, 70.60frames/s] 
  Detected: 41 | Progress: 100%|| 2457/2457 [00:36<00:00, 67.27frames/s] 
100%|| 90/90 [00:11<00:00,  7.58images/s] 
  Detected: 24 | Progress: 100%|| 3603/3603 [00:53<00:00, 67.30frames/s] 
  7%|██████████████▌                                                                                                                                                                                                            | 5/75 [00:00<00:11,  6.27images/s]'bool' object has no attribute 'shape'████████████████████████████████| 116/126 [00:16<00:01,  8.30images/s] 
 97%|▏     | 73/75 [00:10<00:00,  7.64images/s]'bool' object has no attribute 'shape'████████████████████████████████████▉                                                                                                                                                   | 696/2181 [00:20<00:44, 33.42frame/s] 
100%|| 2181/2181 [00:33<00:00, 64.84frame/s]
100%|| 2181/2181 [00:33<00:00, 110.37frame/s]
```

====================================================================================
Code
-----------------------
```
def split_video_into_scenes(video_path, threshold, with_html=None):
    video = open_video(video_path) 
    video_name = os.path.basename(video_path)
    video_name_without_ext = os.path.splitext(video_name)[0]
    video_folder = os.path.dirname(video_path)
    output_folder = os.path.join(video_folder, video_name_without_ext)
    if not os.path.exists(output_folder):
        os.makedirs(output_folder)
        
    html_folder = os.path.join(output_folder, 'html')
    if not os.path.exists(html_folder):
        os.makedirs(html_folder)
    
    scene_manager = SceneManager()
    scene_manager.add_detector(
        ContentDetector(threshold=threshold))
    scene_manager.detect_scenes(video, show_progress=True)
    scene_list = scene_manager.get_scene_list()
    if with_html:
        image_list = sm.save_images(scene_list, video, output_dir=html_folder, show_progress=True, scale=0.1)
        sm.write_scene_list_html(os.path.join(html_folder, 'result.html'), scene_list, image_filenames=image_list)
    split_video_ffmpeg(video_path, scene_list, output_folder, show_progress=True)

if __name__ == '__main__':
    video_dir = input('please input a directory\n')
    threshold = 30.0
    with_html = True
    if video_dir.lower().endswith('.mp4'):
        split_video_into_scenes(video_dir, threshold, with_html)
    else:
        video_files = [os.path.join(video_dir, f) for f in os.listdir(video_dir) if f.lower().endswith('.mp4')]
        
        with concurrent.futures.ThreadPoolExecutor(max_workers=4) as executor:
            futures = [executor.submit(split_video_into_scenes, video_path, threshold, with_html) for video_path in video_files]
            for future in concurrent.futures.as_completed(futures):
                try:
                    future.result()
                except Exception as e:
                    print(str(e))

    input('\n finished……')
```

**Reply by Breakthrough** [reply]
Can you try using `video = open_video(video_path, backend="moviepy")` instead at the top of `split_video_into_scenes`?

Edit: Reviewing the code, this error should only be possible from `save_images` and not `split_video_ffmpeg`.

**Reply by Breakthrough** [reply]
I think there might be a logic bug here:

https://github.com/Breakthrough/PySceneDetect/blob/3fb8d8a1a6f01373398cb8c49061a7fb9348a1fa/scenedetect/scene_manager.py#L544

Should read `if frame_im is not False:` instead.

That being said it would result in the image not being written, so I am curious if there is any corruption to this video.  Are you able to share a sample?

**Reply by pengyouak** [reply]
视频尺寸超过了GitHub的附件限制，我需要怎么发上来呢？
How can I upload a video that exceeds the attachment limit on GitHub?

**Reply by Breakthrough** [reply]
> 视频尺寸超过了GitHub的附件限制，我需要怎么发上来呢？ How can I upload a video that exceeds the attachment limit on GitHub?

I don't think you can, how large is the file?  You may have to upload elsewhere, you may also try the Discord or upload to Google Drive.

**Reply by pengyouak** [reply]
[Microsoft OneDrive](https://1drv.ms/v/s!AgnDI1RBqyYUgbMlCPj108WeOkUaAA?e=STPQMW) 试试这个，一共74MB

**Reply by Breakthrough** [reply]
Thanks, yes this is a bug and will be fixed per #455.  Note that you can use different backend like PyAV or MoviePy to avoid this issue in the meantime, thanks!

**Reply by pengyouak** [reply]
非常感谢，我试试PyAV，看起来也不错
